src/systemd.rs

```diff
@@ -53,17 +56,22 @@ fn is_active(service: &str) -> Result<bool> {
     Ok(output.status.code().unwrap() == 0)
 }
 
-fn wait_for(service: &str, state: bool) -> Result<()> {
+fn wait_for(service: &str, wanted_active: bool) -> Result<()> {
     for _ in 0..20 {
-        if state == is_active(service)? {
-            return Ok(());
-        }
+        match (wanted_active, is_active(service)?) {
+            (true, true) | (false, false) => return Ok(()),
+            (true, false) | (false, true) => (),
+        };
         thread::sleep(Duration::from_millis(50));
     }
-    match state {
-        true => Err(eyre!("Time out waiting for activation")),
-        false => Err(eyre!("Time out waiting for deactivation")),
-    }
+
+    Err(eyre!(
+        "Time out waiting for {}",
+        match wanted_active {
+            true => "activation",
+            false => "deactivation",
+        }
```

Comment:
I'm with clippy here, this should be an if-else.

Reply:
good catch